### PR TITLE
[Merged by Bors] - feat: convergence of infinite sums in nonarchimedean abelian groups

### PR DESCRIPTION
**Pull request by trivial1711**
Let `G` be a complete nonarchimedean abelian group and let `f : α → G` be a function. We prove that `f` is unconditionally summable if and only if `f a` tends to zero on the cofinite filter on `α`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by trivial1711** [comment]
One question I have for the reviewers is what folder this file should be in (`Mathlib.Topology.Algebra.InfiniteSum.Nonarchimedean`, or `Mathlib.Topology.Algebra.Nonarchimedean.InfiniteSum`?) . The other is what namespace the theorems should be in (`NonarchimedeanAddGroup.summable_iff_tendsto_cofinite_zero`, or `summable_iff_tendsto_cofinite_zero`?).

**Review by fpvandoorn — COMMENTED** [review]
I don't have time for a full review at the moment, but here is one comment already.

The choice of file name and namespace seems fine to me (maybe not in a namespace might be ok, but this namespace is fine if there are already other lemmas in it).

File: Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+namespace NonarchimedeanAddGroup
+
+open scoped Pointwise
+
+/-- Let `G` be a nonarchimedean abelian group, and let `f : α → G` be a function that tends to
+zero on the filter of cofinite sets. For each finite subset of `α`, consider the partial sum of `f`
+on that subset. These partial sums form a Cauchy filter. -/
+theorem cauchy_partial_sums_of_tendsto_cofinite_zero {α G : Type*} [AddCommGroup G] [UniformSpace G]
+    [UniformAddGroup G] [NonarchimedeanAddGroup G] [DecidableEq α] {f : α → G}
+    (hf : Tendsto f cofinite (𝓝 0)) : Cauchy (map (fun S ↦ Finset.sum S f) atTop) := by
```
[the remark this answers is not in the document]
Comment (trivial1711) [reply Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean:27]: Thanks! I'll keep this in mind.

**Comment by trivial1711** [comment]
I made this proof more complicated than it needs to be. I can simplify it by using `cauchySeq_finset_iff_tsum_vanishing`. I will work on this.

File: Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  that for any `t : Finset α` disjoint from `s`, we have `∏ i in t, f i ∈ U`. -/
+  apply cauchySeq_finset_iff_prod_vanishing.mpr
+  intro U hU
+
+  -- Since `G` is nonarchimedean, `U` contains an open subgroup `V`.
+  rcases is_nonarchimedean U hU with ⟨V, hV⟩
+
+  /- Let `s` be the set of all indices `i : α` such that `f i ∉ V`. By our assumption `hf`, this is
+  finite. -/
+  use (tendsto_def.mp hf V (OpenSubgroup.mem_nhds_one V)).toFinset
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean:43]:
```suggestion
  use (tendsto_def.mp hf V V.mem_nhds_one).toFinset
```

File: Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+that tends to zero on the filter of cofinite sets. For each finite subset of `α`, consider the
+partial sum of `f` on that subset. These partial sums form a Cauchy filter."]
+theorem cauchy_partial_prod_of_tendsto_cofinite_one {α G : Type*} [CommGroup G] [UniformSpace G]
+    [UniformGroup G] [NonarchimedeanGroup G] {f : α → G}
+    (hf : Tendsto f cofinite (𝓝 1)) : Cauchy (map (fun s ↦ ∏ i in s, f i) atTop) := by
+  /- Let `U` be a neighborhood of `1`. It suffices to show that there exists `s : Finset α` such
+  that for any `t : Finset α` disjoint from `s`, we have `∏ i in t, f i ∈ U`. -/
+  apply cauchySeq_finset_iff_prod_vanishing.mpr
+  intro U hU
+
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean:37]:
```suggestion
```
Please remove blank lines in proofs (also below).

**Review by fpvandoorn — COMMENTED** [review]
This looks very clean! LGTM, just two tiny style suggestions.

**Comment by trivial1711** [comment]
Thank you for the review. I have made the changes you suggest.

File: Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+namespace NonarchimedeanGroup
+
+/--Let `G` be a nonarchimedean multiplicative abelian group, and let `f : α → G` be a function that
+tends to one on the filter of cofinite sets. For each finite subset of `α`, consider the partial
+product of `f` on that subset. These partial products form a Cauchy filter. -/
+@[to_additive "Let `G` be a nonarchimedean additive abelian group, and let `f : α → G` be a function
+that tends to zero on the filter of cofinite sets. For each finite subset of `α`, consider the
+partial sum of `f` on that subset. These partial sums form a Cauchy filter."]
+theorem cauchy_partial_prod_of_tendsto_cofinite_one {α G : Type*} [CommGroup G] [UniformSpace G]
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean:28]:
```suggestion
theorem cauchySeq_prod_of_tendsto_cofinite_one {α G : Type*} [CommGroup G] [UniformSpace G]
```
I like the new formulation using `CauchySeq`, please also update the name (and you can remove `_partial` from the name).

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**